index.js

```diff
@@ -0,0 +1,134 @@
+document.addEventListener('DOMContentLoaded', () => {
+  const operatorBtn = document.querySelectorAll('.operator'),
+    numberBtn = document.querySelectorAll('.number'),
+    pointBtn = document.getElementById('point'),
+    clearBtn = document.querySelectorAll('.clear-btn'),
+    display = document.getElementById('display')
+  let currValue = 0,
+    currentOperator = '',
+    isNewNumber = false
+
+  const OPERATORS = {
+    plus: '+',
+    minus: '-',
+    multiply: '*',
+    devide: '/',
+    equals: '=',
+  }
+
+  const arrOper = Object.values(OPERATORS)
+  const pressOperation = (el) => {
+    let localMemory = display.value
+    if (isNewNumber && currentOperator !== '=') {
+      display.value = currValue
+    } else {
+      isNewNumber = true
+      switch (currentOperator) {
+        case '+':
+          currValue += +localMemory
+          break
+        case '-':
+          currValue -= +localMemory
+          break
+        case '*':
+          currValue *= +localMemory
+          break
+        case '/':
+          currValue /= +localMemory
+          break
+        default:
+          currValue = +localMemory
+      }
+      display.value = currValue
+    }
+    currentOperator = el
+  }
+
+  operatorBtn.forEach((el) =>
+    el.addEventListener('click', (e) => {
+      pressOperation(e.target.textContent)
+    })
+  )
+
+  const pressNum = (el) => {
+    if (isNewNumber) {
+      display.value = el
+      isNewNumber = false
+    } else {
+      if (display.value === '0') display.value = el
+      else display.value += el
+    }
+  }
+
+  numberBtn.forEach((el) =>
+    el.addEventListener('click', (e) => {
+      pressNum(e.target.textContent)
+    })
+  )
+
+  const KEYS = {
+    C: 'C',
+    CE: 'CE',
+  }
+
+  const pressClearBtn = (el) => {
+    if (el === KEYS.C) {
+      display.value = '0'
+      currValue = 0
+      isNewNumber = true
+      currentOperator = ''
+    } else if (el === KEYS.CE) {
+      display.value = display.value.length > 1 ? display.value.slice(0, -1) : ''
+     if (isNewNumber === true) {
+       currValue = +display.value
+       isNewNumber = true
+     }
```

Comment:
- Please, refactor the condition here;
- the logic is not clear here: if isNewNumber is true -> set isNewNumber = true. It makes no sense. Check the logic here and if the logic is correct. remove the assignment to variable isNewNumber.

Reply:
I did it, but I don't know why I didn't save it)